### PR TITLE
Simplify state conversions (excludes Crystal environments)

### PR DESCRIPTION
**Pull request by alexhernandezgarcia**
This PR simplifies state conversions in the following ways:

- `state*2oracle` has been removed as well as other mentions to `oracle`. In short, references to the oracle are now unified into references to the proxy.
- `statebatch2proxy` and `statetorch2proxy` are now unified by `states2proxy`
- `statebatch2policy` and `statetorch2policy` are now unified by `states2policy`
- `state2proxy` has been removed from all environments because it is enough with the implementation in the base environment, which calls `states2proxy`.
- `policy2state` has been removed.

I have also taken the chance to add richer docstrings. I believe all other changes should be related to the above.

- [x] I have run the sanity checks and everything seems fine.
- [x] Tests are passed locally.

Note: the PR includes some deletion of files, which come from a different branch because I messed up things somehow... :sweat: Sorry for that!

### CODE REVIEW

**Comment by alexhernandezgarcia** [comment]
@carriepl I would like to have your advice on how to proceed with this PR. The situation is the following:
- This branch was off `main` before `main` included the crystal's updates.
- Therefore, this PR does not update the crystal's environments.
- Therefore, if it is merged to `main`, then `main` will be broken, until the changes updating the crystal's environments are applied.

I am thinking that an auxiliary branch off `main` would do, but I wonder if there is any other better way.

**Comment by carriepl** [comment]
> @carriepl I would like to have your advice on how to proceed with this PR. The situation is the following:
> 
> * This branch was off `main` before `main` included the crystal's updates.
> * Therefore, this PR does not update the crystal's environments.
> * Therefore, if it is merged to `main`, then `main` will be broken, until the changes updating the crystal's environments are applied.
> 
> I am thinking that an auxiliary branch off `main` would do, but I wonder if there is any other better way.

Is there any obstacle to rebasing this branch (or a copy of this branch, to be safer) over the main and then updating the crystal environments in the rebased version of this branch?

**Comment by alexhernandezgarcia** [comment]
> > @carriepl I would like to have your advice on how to proceed with this PR. The situation is the following:
> > 
> > * This branch was off `main` before `main` included the crystal's updates.
> > * Therefore, this PR does not update the crystal's environments.
> > * Therefore, if it is merged to `main`, then `main` will be broken, until the changes updating the crystal's environments are applied.
> > 
> > I am thinking that an auxiliary branch off `main` would do, but I wonder if there is any other better way.
> 
> Is there any obstacle to rebasing this branch (or a copy of this branch, to be safer) over the main and then updating the crystal environments in the rebased version of this branch?

That's about what I thought you could say. The only obstacle is that I don't feel super at ease with rebasing yet. Would you mind outlining what the steps should be?

File: gflownet/utils/common.py [hunk 1]
```diff
@@ -167,36 +167,36 @@ def batch_with_rest(start, stop, step, tensor=False):
...
-        return x.type(torch.long).to(device)
+        return x.to(device=device, dtype=torch.long)
     else:
         return torch.tensor(x, dtype=torch.long, device=device)
 
 
 def tint(x, device, int_type):
     if isinstance(x, list) and torch.is_tensor(x[0]):
-        return torch.stack(x).type(int_type).to(device)
+        return torch.stack(x).to(device=device, dtype=int_type)
```
[the remark this answers is not in the document]
Comment (alexhernandezgarcia) [reply gflownet/utils/common.py:188]: You're right. Rather for another PR indeed.

**Comment by alexhernandezgarcia** [comment]
So I am doing this the way it is easiest and safest to me.

1. Create auxiliary branch off current main (`conversions-simplify-merge`).
2. Resolve conflicts and merge this PR into `conversions-simplify-merge` (copy of `main`)
3. Implement rest of changes related to the simplification of state conversions (on crystal environments) in a new branch off `conversions-simplify-merge` and merge into `main` in a new PR (to be done).

**Pull request merged**